util/optical_flow.py

```diff
@@ -0,0 +1,147 @@
+
+            # Get features to track
+            p0 = cv2.goodFeaturesToTrack(
+                gray1, mask=None, **feature_params)
+
+            # Get second frame
+            frame2 = cv2.imread(image_files[rand_frame+1])
+            gray2 = cv2.cvtColor(frame2, cv2.COLOR_BGR2GRAY)
+
+            # Calculate optical flow
+            p1, st, err = cv2.calcOpticalFlowPyrLK(
+                gray1, gray2, p0, None, **lk_params)
+
+            # Select good points
+            good_new = p1[st == 1]
+            good_old = p0[st == 1]
+
+            # Draw the tracks
+            dy = 0
+            dx = 0
+            for i, (new, old) in enumerate(zip(good_new,
+                                            good_old)):
+                a, b = new.ravel()
+                c, d = old.ravel()
+                dx = dx + c-a
+                dy = dy + d-b
+            if (len(good_new) <= 0):
+                print("No features found in frame ", rand_frame)
+                continue
+            Dx.append(dx/len(good_new))
+            Dy.append(dy/len(good_new))
+
+        # Do classification
+        DxDyRatios = []
+        for i in range(0, len(Dx)):
+            DxDyRatios.append(abs(Dx[i]/Dy[i]))
+
+        # Check for silent failure
+        if (len(DxDyRatios) == 0):
+            print("Camera mount checking failed.")
+            exit
+
+        # Classify camera mount
+        if (statistics.median(DxDyRatios) > threshold_dxdy_ratio):
+            if (statistics.median(Dx) < 0.0):
+                print("Passenger side camera orientation")
+            else:
+                print("Driver side camera orientation")
+        else:
+            print("Forward or backward facing camera orientation")
+
+        print("FINISHED!")
+    else:
+        print("Less than 5 frames. Skipping optical flow based camera mount classificatin.")
+        print("FINISHED!")
+
+
+def list_image_files(directory: str, unzip=False):
+    """
+    List all JPEG image files in a directory.
+
+    If 'unzip' is True and the directory is a zip file, it extracts the contents.
+
+    Parameters:
+        directory (str): Path to the directory or zip file.
+        unzip (bool, optional): Extract zip file if True. Defaults to False.
+
+    Returns:
+        list: List of JPEG image file paths.
+
+    Example:
+        >>> list_image_files("/path/to/images", unzip=True)
+        ['/path/to/images/image1.jpg', '/path/to/images/image2.jpg', ...]
+    """
+    image_files = []
+    if unzip and directory.endswith(".zip"):
+        with zipfile.ZipFile(directory, 'r') as zip_ref:
+            zip_ref.extractall(os.path.dirname(directory))
+        directory = os.path.splitext(directory)[0]  # Update directory to the extracted folder
+    for filename in os.listdir(directory):
+        if filename.endswith(".jpg"):
+            image_files.append(os.path.join(directory, filename))
+    return sorted(image_files, key=lambda x: int(x.split('/')[-1].split('.')[0]))
```

Comment:
can you add a `try` `except` here to first try to sort by coerced int like you have here, then default to lexicographic (default sort behavior) in case that throws an error, e.g.,

```
In [6]: int("hello")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[6], line 1
----> 1 int("hello")

ValueError: invalid literal for int() with base 10: 'hello'
```